mongosql/src/algebrizer/test.rs

```diff
@@ -5667,20 +5667,97 @@ mod select_clause {
     }
 
     test_algebrize!(
-        select_distinct_not_allowed,
+        select_values_distinct,
```

Comment:
apologies if I'm missing it but I think a test with the `NonStarStandardSelectBody` Error could be useful as well, here, in the spec tests, or both. Codifying that we dont allow stuff like `select distinct (foo.*, bar.a)` or (I think) even `select distinct foo.*`

Reply:
I added an algebrizer e2e test to check that `SELECT DISTINCT *, a` gets that error.  A query like `select distinct foo.*` will go down the `ast::SelectBody::Values(exprs)` branch and works ok.